rpcs3/Emu/Cell/Modules/cellGem.cpp

```diff
@@ -264,6 +264,37 @@ struct gem_config_data
 
 	u64 start_timestamp_us = 0;
 
+	atomic_t<u32> m_wake_up = 0;
+	atomic_t<u32> m_done = 0;
+
+	void wake_up()
+	{
+		m_wake_up.release(1);
+		m_wake_up.notify_one();
+	}
+
+	void done()
+	{
+		m_done.release(1);
+		m_done.notify_one();
+	}
+
+	void wait_for_result()
+	{
+		if (!m_done)
```

Comment:
Should be a while loop, code should not assume that wait is accurate and won't wake up spuriously